packages/extension/src/companion/useExtensionPermission.ts

```diff
@@ -27,12 +27,14 @@ export const getContentScriptPermission = (): Promise<boolean> =>
     origins: ['*://*/*'],
   });
 
+let hasInjectedScripts = false;
```

Comment:
Can't we use useState here?

Reply:
Good point, though the function is not a hook as it is also needed to run in the background worker, so we can't turn it to state.